onnxruntime/core/framework/session_options.h

```diff
@@ -110,6 +110,10 @@ struct SessionOptions {
   // Deterministic compute is likely not as performant. This option is default to false.
   bool use_deterministic_compute = false;
 
+  // By default the session uses fix memory size (32M) for Conv algo search, the final algo might not be the best.
+  // If this is set to true, try to use as much as possible memory for algo search.
+  bool use_more_mem_for_conv = false;
```

Comment:
This option doesn't apply to CPU kernel. 
I think we should follow cudnn_conv_algo_search's pattern, and has it as a config in CUDAExecutionProviderInfo.

Reply:
also, let's rename this for a more explicit name, e.g. cudnn_conv_use_full_workspace